src/type_checker/checker.rs

```diff
@@ -365,6 +366,7 @@ impl<B: Backend> TypeChecker<B> {
                 let idx_typ = self.compute_type(idx, typed_fn_env)?;
                 match idx_typ.map(|t| t.typ) {
                     Some(TyKind::BigInt) => (),
+                    Some(TyKind::Generic(_)) => (),
```

Comment:
it feels weird to have code like that, but I guess Rust seem to allow that (https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2a95c19e8da557dc1bddf8bb667baa7d)

but it doesn't allow something like `a[NN+1]` (as usual no operations on generic parameter in Rust).

I think these two points should be added to the RFC

Reply:
It is useful to have something like a[NN - 1] to return the last item in generic way, no?

Reply:
At least for this example we should have builtins like a[-1]

Reply:
TyKind::Generic is removed in another PR